katz/src/main/kotlin/katz/Option.kt

```diff
@@ -134,3 +136,23 @@ sealed class Option<out A> {
  * @param default  the default expression.
  */
 fun <B> Option<B>.getOrElse(default: () -> B): B = fold({ default() }, { it })
+
+/**
+ * Companion interface for [Option] type
+ * outside due to https://youtrack.jetbrains.com/issue/KT-10532
+ */
+private interface OptionInstances {
+    fun <A> HK<Option.F, A>.ev(): Option<A> = this as Option<A>
+
+    fun monad() = object : Monad<Option.F> {
+        override fun <A, B> map(fa: HK<Option.F, A>, f: (A) -> B): HK<Option.F, B> =
+                fa.ev().map(f)
+
+        override fun <A> pure(a: A): HK<Option.F, A> = Option.Some(a)
```

Comment:
what do you think about use `fromNullable()` inside `pure()` instead of call directly Option.Some() ?? for that this PR need to be Rebased with master